install/ui/src/freeipa/plugins/certmap.js

```diff
@@ -0,0 +1,261 @@
+                        },
+                        'ipacertmapissuer',
+                        'ipacertmapmaprule',
+                        'ipacertmapmatchrule',
+                        {
+                            $type: 'multivalued',
+                            name: 'associateddomain'
+                        },
+                        'ipacertmappriority'
+                    ]
+                }
+            ]
+        }
+    ],
+    adder_dialog: {
+        fields: [
+            'cn',
+            {
+                $type: 'checkbox',
+                name: 'ipaenabledflag',
+                checked: 'checked'
+            },
+            {
+                $type: 'textarea',
+                name: 'description'
+            },
+            'ipacertmapissuer',
+            'ipacertmapmaprule',
+            'ipacertmapmatchrule',
+            {
+                $type: 'multivalued',
+                name: 'associateddomain'
+            },
+            'ipacertmappriority'
+        ]
+    }
+};};
+
+
+var make_certmapconfig_spec = function() {
+return {
+    name: 'certmapconfig',
+    defines_key: false,
+    facets: [
+        {
+            $type: 'details',
+            facet_groups: [certmap.search_facet_group],
+            facet_group: 'search',
+            sections: [
+                {
+                    name: 'details',
+                    fields: [
+                        {
+                            $type: 'checkbox',
+                            name: 'ipacertmappromptusername'
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+};};
+
+
+/**
+ * Multivalued widget which is used for working with user's certmap.
+ *
+ * @class
+ * @extends IPA.custom_command_multivalued_widget
+ */
+certmap.certmap_multivalued_widget = function (spec) {
+
+    spec = spec || {};
+
+    spec.adder_dialog_spec = spec.adder_dialog_spec || {
+        title: '@i18n:objects.certmap.adder_title',
+        fields: [
+            {
+                $type: 'textarea',
+                name: 'ipacertmapdata',
+                label: '@i18n:objects.certmap.data_label'
+            }
+        ]
+    };
+
+    var that = IPA.custom_command_multivalued_widget(spec);
+
+    that.create_remove_dialog_title = function(row) {
+        return text.get('@i18n:objects.certmap.deleter_title');
+    };
+
+    that.create_remove_dialog_message = function(row) {
+        var message = text.get('@i18n:objects.certmap.deleter_content');
+        message = message.replace('${data}', row.widget.new_value);
+
+        return message;
+    };
+
+    /**
+     * Compose options for add command.
+     * @return {Object} options
+     */
+    that.create_add_options = function() {
+        var options = {};
+        var certmapdata = that.adder_dialog.get_field('ipacertmapdata').value;
+
+        options['ipacertmapdata'] = certmapdata;
+
+        return options;
+    };
+
```

Comment:
@dkupka @flo-renaud I would say that it would be kind of hard (especially in CLI) for users to understand which options should be used together and which can't be used together. From point of view of WebUI, there is no problem to create dialog, which allows only correct combinations of options.